aten/src/ATen/native/Blas.cpp

```diff
@@ -91,4 +94,31 @@ Tensor mv(const Tensor &self, const Tensor &vec) {
   return native::mv_out(result, self, vec);
 }
 
+Tensor dot(const Tensor &self, const Tensor &other){
+  at::NoNamesGuard guard;
+
+  TORCH_CHECK(
+      self.dim() == 1 && other.dim() == 1,
+      "1D tensors expected, got, ",
+      self.dim(), ", ",
+      other.dim(),
+      " tensors");
+
+  TORCH_CHECK(
+      self.numel() == other.numel(),
+      "inconsistent tensor size, expected tensor [",
+      self.numel(),
+      "] and src [",
+      other.numel(), "] to have the same number of elements, but got ",
+      self.numel(), " and ",
+      other.numel(),
+      " elements respectively");
+
+  return AT_DISPATCH_ALL_TYPES(self.scalar_type(), "dot", [&] {
+    Tensor result = at::empty({}, self.options());
+    result.fill_(dot_impl<scalar_t>(self.numel(), self.data_ptr<scalar_t>(), self.stride(0), other.data_ptr<scalar_t>(), other.stride(0)));
+    return result;
+  });
```

Comment:
Would be great if someone can tell me the equivalent of this machinery in ATen

https://github.com/pytorch/pytorch/blob/3bbedb34b9b316729a27e793d94488b574e1577a/aten/src/TH/generic/THTensorEvenMoreMath.cpp#L87-L94

Reply:
Why do we need this machinery instead of directly calling THBlas_(dot) with correct sizes and strides?

Reply:
I didn't have the time to verify, but the comment 
```
/* we use a trick here. careful with that. */ 
```

leads me to be believe that, with `TH_TENSOR_APPLY2`, Could be doing `a[:50].dot(b[:50]) + a[50:].dot(b[50:])` (more reference in the below snippet) in parallel for larger vectors, which might give better performance over directly calling the function. 

I am yet to benchmark the changes. So that might shed some light on it.

```python
>>> import numpy as np
>>> a = np.random.randn(100)
>>> b = np.random.randn(100)
>>> a[:50].dot(b[:50]) + a[50:].dot(b[50:])
15.157316408439492
>>> a.dot(b)
15.157316408439494
>>> 
```

Reply:
I was wrong it doesn't do that. My bad.